### PR TITLE
feat(ui): Use cookies in the UI. Closes #1949

### LINKED ISSUE #1949: TODO - we should not put the token in the URL - OSWAP obvs
in `artifact_server.go` line 94 we should not get the token from the URL query parameter because this would appear in HTTP access logs and is a security risk. 

We can use a cookie instead, but only if the bearer token in under 4KB in size.

Alternatively, can we create a GPRC endpoint somehow?

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
* [x] Optional. I've added My organization is added to the README.
* [x] I've signed the CLA and required builds are green. 

 Closes #1949

This PR allows you to authenticate using either the Authorization header, or a cookie. This means we can download artifacts without have the token in the URL, which is not secure. 

Basically, if you are using the API you should have `Authorization: Bearer ...`, if you are the UI should should have `Cookie: authorization=...`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -339,6 +339,8 @@ up:
 	kubectl -n argo scale deployment/argo-server --replicas 1
 	# Wait for pods to be ready
 	kubectl -n argo wait --for=condition=Ready pod --all -l app --timeout 2m
+	# Token
+	kubectl -n argo get `kubectl -n argo get secret -o name | grep argo-server` -o jsonpath='{.data.token}' | base64 -d
```
Comment (alexec) [inline Makefile]: I want to know what the login token is!

Comment (simster7) [reply Makefile]: Would it be a security concern that the secret gets sent to stdout?

Comment (alexec) [reply Makefile]: for tests only
